From 37a1a6118d6cb95e3729e2bf81165093cad87a11 Mon Sep 17 00:00:00 2001
From: Mark Enriquez <enriquezmark36@gmail.com>
Date: Mon, 23 Mar 2020 13:24:54 +0800
Subject: [PATCH 1/8] libcamera: kanas: scx15: Enable s5k4ec mipi sensor, move
 HI702 to index 2

This was disabled due to 5066fc1885ca313bd868ab9fbc82410542618d78
(camera: scx15: fix compile for N)

But, kanas3g phone needs this. Will fix the s5k4ecgx mipi sensor
to feature parity later on.

TL;DR; When switching over to the stock or opensource impl of SPRD libcamera HAL
delete the "/data/misc/media/sensor.file"

And, to point it out, that specific change could make the cached
identification to be invalid if ever someone did used the stock libcamera
HAL before this opensource one. The HAL seems to never clear that cache so
that it will persist until the user manually removes the file at:
/data/misc/media/sensor.file

The guys at SPRD realized that doing an identify to all of that
other sensors (the ones that the change commented ou) will take some time
and it has to do it every time the HAL is init'd (or every time you use
either of the camera). So they did it once and saved the results to
a file, once.

This stale cache may cause the identifier to fail to load the driver
and may even fail to power it down when initiallizing the other sensor
which may also make that other sensor fail at init.

Change-Id: Ifdb85b867d2afdb697e094f3f52121b1b16c6aea
Signed-off-by: Mark Enriquez <enriquezmark36@gmail.com>
---
 libcamera/scx15/Camera.mk               |  1 +
 libcamera/scx15/Camera2.mk              |  1 +
 libcamera/scx15/sc8830/src/sensor_cfg.c | 12 ++++++++++--
 3 files changed, 12 insertions(+), 2 deletions(-)

diff --git a/libcamera/scx15/Camera.mk b/libcamera/scx15/Camera.mk
index 263ed75..f9f1258 100644
--- a/libcamera/scx15/Camera.mk
+++ b/libcamera/scx15/Camera.mk
@@ -64,6 +64,7 @@ LOCAL_SRC_FILES:= \
 	sensor/sensor_hi702_ccir.c \
 	sensor/sensor_pattern.c \
 	sensor/sensor_sr352.c \
+	sensor/sensor_s5k4ecgx_mipi.c \
 	vsp/sc8830/src/jpg_drv_sc8830.c \
 	jpeg/jpeg_fw_8830/src/jpegcodec_bufmgr.c \
 	jpeg/jpeg_fw_8830/src/jpegcodec_global.c \
diff --git a/libcamera/scx15/Camera2.mk b/libcamera/scx15/Camera2.mk
index a60b008..9a96baa 100644
--- a/libcamera/scx15/Camera2.mk
+++ b/libcamera/scx15/Camera2.mk
@@ -64,6 +64,7 @@ LOCAL_SRC_FILES:= \
 	sensor/sensor_hi702_ccir.c \
 	sensor/sensor_pattern.c \
 	sensor/sensor_sr352.c \
+	sensor/sensor_s5k4ecgx_mipi.c \
 	vsp/sc8830/src/jpg_drv_sc8830.c \
 	jpeg/jpeg_fw_8830/src/jpegcodec_bufmgr.c \
 	jpeg/jpeg_fw_8830/src/jpegcodec_global.c \
diff --git a/libcamera/scx15/sc8830/src/sensor_cfg.c b/libcamera/scx15/sc8830/src/sensor_cfg.c
index d6100f5..2672522 100644
--- a/libcamera/scx15/sc8830/src/sensor_cfg.c
+++ b/libcamera/scx15/sc8830/src/sensor_cfg.c
@@ -52,7 +52,7 @@ extern SENSOR_INFO_T g_ov8830_mipi_raw_infoextern;
 //extern SENSOR_INFO_T g_imx179_mipi_raw_info;
 //extern SENSOR_INFO_T g_ov8865_mipi_raw_info;
 //extern SENSOR_INFO_T g_ov13850_mipi_raw_info;
-//extern SENSOR_INFO_T g_s5k4ec_mipi_yuv_info;
+extern SENSOR_INFO_T g_s5k4ec_mipi_yuv_info;
 extern SENSOR_INFO_T g_HI702_yuv_info;
 //extern SENSOR_INFO_T g_ov5640_yuv_info;
 //extern SENSOR_INFO_T g_OV7675_yuv_info;
@@ -76,7 +76,7 @@ extern SENSOR_INFO_T g_sr352_yuv_info;
 const SENSOR_INFO_T* main_sensor_infor_tab[]=
 {
 #ifdef CONFIG_BACK_CAMERA_MIPI
-	//&g_s5k4ec_mipi_yuv_info,
+	&g_s5k4ec_mipi_yuv_info,
 	//&g_ov8825_mipi_raw_info,
 	//&g_ov8830_mipi_raw_info,
 	//&g_ov5640_mipi_yuv_info,
@@ -111,6 +111,14 @@ const SENSOR_INFO_T* sub_sensor_infor_tab[]=
 	//&g_OV7675_yuv_info,
 	//&g_GC0308_yuv_info,
 	//&g_GC2035_yuv_info,
+
+	/*
+	 * kanas changes: Since we want to use the cache the vendor
+	 * blob has already created, we need the make sure the
+	 * HI702 sensor is at the third position (index 2)
+	 */
+	&g_HI702_yuv_info,
+	&g_HI702_yuv_info,
 	&g_HI702_yuv_info,
 	#if defined(CONFIG_CAMERA_X3542)
 	//&g_GC0311_yuv_info,
-- 
2.25.0

